x-pack/legacy/plugins/epm/server/lib/kibana/index_pattern/install.test.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import path from 'path';
+import { readFileSync } from 'fs';
+import { safeLoad } from 'js-yaml';
+import { flattenFields, dedupFields } from './install';
+import { Field } from '../../fields/field';
+
+// Add our own serialiser to just do JSON.stringify
+expect.addSnapshotSerializer({
+  print(val) {
+    return JSON.stringify(val, null, 2);
+  },
+
+  test(val) {
+    return val;
+  },
+});
+
+const loadYamlFile = (filePath: string) => {
+  const ymlPath = path.join(__dirname, filePath);
+  const fieldsYML = readFileSync(ymlPath, 'utf-8');
+  const fields: Field[] = safeLoad(fieldsYML);
+  return fields;
+};
+const fields = loadYamlFile('./tests/nginx.fields.yml');
```

Comment:
I would suggest for the tests, to just iterate through all *.yml files in this directory. This makes adding a new tests super simple and you tests all of them for dedup and flatten.